images/dvcr-artifact/cmd/dvcr-cleaner/cmd/ls.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	NamespaceFlag     string
+	AllImagesFlag     bool
+	AllNamespacesFlag bool
+	ImageRepoDir      = "/var/lib/registry/docker/registry/v2/repositories"
+)
+
+var LsCmd = &cobra.Command{
+	Use:   "ls",
+	Short: "List of virtual images",
+	Args:  cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+}
+
+var lsViCmd = &cobra.Command{
+	Use:   "vi [imageName]",
+	Short: "Get virtual image or list of images",
+	Args: cobra.MatchAll(
+		cobra.MaximumNArgs(1),
+		cobra.OnlyValidArgs,
+	),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 && !AllImagesFlag && !AllNamespacesFlag {
+			return cmd.Help()
+		}
+
+		if len(args) != 0 && AllImagesFlag {
+			return errors.New("flag `all` cannot be used with `imageName`")
+		}
+
+		if len(args) != 0 {
+			image := args[0]
+			info, err := os.Stat(fmt.Sprintf("%s/vi/%s/%s", ImageRepoDir, NamespaceFlag, image))
+			if errors.Is(err, os.ErrNotExist) {
+				return fmt.Errorf("virtual image %q not found in %q namespace", image, NamespaceFlag)
+			}
+			if err != nil {
+				return fmt.Errorf("cannot get virtual image %q in %q namespace: %w", image, NamespaceFlag, err)
+			}
+			fmt.Println(info.Name())
+			return nil
+		}
+
+		if AllImagesFlag {
+			images, err := ListVirtualImages(NamespaceFlag)
+			if err != nil {
+				return err
+			}
+			fmt.Println(strings.Join(images, "\n"))
+			return nil
+		}
+
+		if AllNamespacesFlag {
```

Comment:
I think this flag should have higher priority than AllImagesFlag.

Reply:
`all` and `all-namespaces` cannot be used at the same time.
https://github.com/deckhouse/virtualization/blob/57fb97971074beab9d12b9c932ab77e4468fd611/images/dvcr-artifact/cmd/dvcr-cleaner/cmd/ls.go#L197